test/compiler.test.js

```diff
@@ -1096,6 +1191,55 @@ describe('compiler', function() {
         'sourceFile', moduleJS);
     });
 
+    it('loads middleware from coffeescript', function() {
+      var coffee = appdir.writeFileSync('my-middleware.coffee', '');
+      appdir.writeConfigFileSync('middleware.json', {
+        'routes': {
+          './my-middleware': {}
+        }
+      });
+
+      var instructions = boot.compile(appdir.PATH);
+
+      expect(instructions.middleware.middleware[0]).have.property(
+        'sourceFile', coffee);
+    });
+
+    it('prefers coffeescript over json for relative path middleware',
+      function() {
+      var coffee = appdir.writeFileSync('my-middleware.coffee', '');
+      appdir.writeFileSync('my-middleware.json', '');
+      appdir.writeConfigFileSync('middleware.json', {
+        'routes': {
+          './my-middleware': {}
+        }
+      });
+
+      var instructions = boot.compile(appdir.PATH);
+
+      expect(instructions.middleware.middleware).to.have.length(1);
+      expect(instructions.middleware.middleware[0]).have.property(
+        'sourceFile', coffee);
+    });
+
+    it('prefers coffeescript over json for non-relative path middleware',
```

Comment:
Please rename to "... for module relative middleware path" to keep this consistent with other test names.